src/thunderbots/software/util/canvas_messenger/canvas_messenger.h

```diff
@@ -14,51 +14,104 @@
 #include <chrono>
 #include <map>
 #include <memory>
+#include <mutex>
 #include <string>
 #include <vector>
 
+#include "ai/world/ball.h"
+#include "ai/world/field.h"
+#include "ai/world/robot.h"
+#include "ai/world/world.h"
 #include "thunderbots_msgs/CanvasLayer.h"
 #include "util/constants.h"
 
 namespace Util
 {
+    /**
+     * This class provides an interface for drawing on the "Canvas" area of the visualizer
+     *
+     * All public methods are (and must be) thread safe, so please be careful when
+     * exposing more public functionality
+     */
     class CanvasMessenger
     {
        public:
+        enum class Layer
+        {
+            STATIC_FEATURES,
+            ROBOTS_AND_BALL
+        };
```

Comment:
I feel strongly about using an enum because it provides compile-time guarantees about what can be used as a layer. 

I'm almost certain we can get the string value of the Enum at compile time (probably with a macro) or alternatively we could implement a quick helper function. Totally happy to implement either of these when we have a reasonable way of sending over layer names to the visualizer.

Reply:
Seems like we could use #268 for this? Then we can use enums but send strings easily?

Reply:
Yep, that would do it (and would probably be the best option).